transports/tcp/src/lib.rs

```diff
@@ -943,8 +985,9 @@ mod tests {
             #[cfg(feature = "async-io")]
             {
                 let (ready_tx, ready_rx) = mpsc::channel(1);
-                let listener = listener::<async_io::Tcp>(addr.clone(), ready_tx);
-                let dialer = dialer::<async_io::Tcp>(addr.clone(), ready_rx);
+                let (port_reuse_tx, port_reuse_rx) = mpsc::channel(1);
```

Comment:
How about using a oneshot here?

https://docs.rs/futures/latest/futures/channel/oneshot/fn.channel.html

Reply:
OK. Updated the commit with a oneshot.